### ISSUE TITLE
UnnecessaryParentheses false positive when using extension function with default parameter

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#927]
I've got the following extension function:

```kotlin
@CheckReturnValue fun <T> Observable<T>.subscribeWithDefault(onNext: (T) -> Unit, onComplete: () -> Unit = {}) =
    subscribe(Consumer(onNext::invoke), RxLogger, Action(onComplete::invoke))
```

I'm calling it like this:

```kotlin
Observable.just(1)
    .subscribeWithDefault({ println(it) })
```

Unfortunatenly I'm forced in this case by the compiler to add the `()`. Detekt thinks they're unnecessary and hence flags this.

This should not be flagged though. I'm using RC7.

**Comment by arturbosch** [comment]
Can this be reproduced by an exaple just with rxjava as a dependency ?

**Comment by vanniktech** [comment]
I don't think the RxJava part is required.
